src/lib/DisposableEventListener.ts

```diff
@@ -17,51 +18,53 @@ import { EventHandler } from 'seng-event/lib/EventDispatcher';
  * @class DisposableEventListener
  */
 
-export default class DisposableEventListener extends Disposable {
-  public eventDispatcher;
-  public type: string;
-  public listener: EventListenerOrEventListenerObject | EventHandler;
-  public useCapture?: boolean;
+export default class DisposableEventListener {
+  public readonly eventDispatcher: EventDispatcher;
+  public readonly type: AddEventListenerParameters[0];
+  public readonly listener: AddEventListenerParameters[1];
+  public readonly options?: AddEventListenerParameters[2];
+
+  private disposed = false;
 
   /**
    * Add an event listener on a HTML element.
    *
-   * @param {NativeEventDispatcher} eventDispatcher the HTML element to listen to.
+   * @param {EventDispatcher} eventDispatcher the HTML element to listen to.
    * @param {string} type the type of event to listen to.
-   * @param {EventListener} listener the method which is called when the eventDispatcher dispatches an event to the
+   * @param {Function} listener the method which is called when the eventDispatcher dispatches an event to the
    * specified type
-   * @param {boolean} useCapture If true, useCapture indicates that the user wishes to initiate capture. After
+   * @param {boolean | EventListenerOptions} options If true, useCapture indicates that the user wishes to initiate capture. After
    * initiating capture, all events of the specified type will be dispatched to the registered listener before being
    * dispatched to any EventTarget beneath it in the DOM tree. Events which are bubbling upward through the tree will
    * not trigger a listener designated to use capture.
    */
   constructor(
-    eventDispatcher,
-    type: string,
-    listener: EventListenerOrEventListenerObject | EventHandler,
-    useCapture?: boolean,
+    eventDispatcher: EventDispatcher,
+    type: AddEventListenerParameters[0],
+    listener: AddEventListenerParameters[1],
+    options?: AddEventListenerParameters[2],
   ) {
-    super();
-
     this.eventDispatcher = eventDispatcher;
     this.type = type;
     this.listener = listener;
-    this.useCapture = useCapture;
+    this.options = options;
```

Comment:
Potentially assigning a legacy `useCapture` boolean to this exposed `options` object might be confusing/wrong if using it that way.

We should either expose it correctly (convert the boolean into the `options` object, or having a separate legacy `useCapture` field), or just make those/that field(s) private.